tools/topology/platform/intel/intel-generic-dmic.m4

```diff
@@ -16,6 +16,12 @@ ifdef(`DMIC_DAI_LINK_48k_NAME',`',define(DMIC_DAI_LINK_48k_NAME, `dmic01'))
 # define(DMIC_DAI_LINK_16k_NAME, `dmic16k')
 ifdef(`DMIC_DAI_LINK_16k_NAME',`',define(DMIC_DAI_LINK_16k_NAME, `dmic16k'))
 
+# Handle possible different channels count for PCM and DAI
```

Comment:
should these changes to dmic-generic and sof-apl-512 be in separate, possible extra commit, before this commit?

Reply:
Yes probably should. I made one commit that contains the topology preparations. There should be a clear no-difference to any built topologies but with all needed preparations step.

Reply:
This is now fixed. I moved the non-TDFB related improvemets to the previous commit.

Reply:
Done!